File: frontend/src/types/role.ts
export interface Role {
  id: number;
  name: string;
  slug: string;
  description: string | null;
  permissions: string[];
  is_active: boolean;
  staff_count: number;
  created_at: string;
}

export interface StaffUser {
  id: number;
  email: string;
  first_name: string;
  last_name: string;
  phone: string | null;
  role: string;
  role_id: number | null;
  role_name: string | null;
  permissions: string[];
  is_active: boolean;
  created_at: string;
  last_login_at: string | null;
}

export const ADMIN_MODULES = [
  'dashboard',
  'products',
  'categories',
  'collections',
  'orders',
  'customers',
  'coupons',
  'inventory',
  'reviews',
  'content',
  'media',
  'analytics',
  'settings',
  'newsletter',
  'contact_messages',
] as const;

export type AdminModule = (typeof ADMIN_MODULES)[number];

export const MODULE_LABELS: Record<AdminModule, string> = {
  dashboard: 'Dashboard',
  products: 'Products',
  categories: 'Categories',
  collections: 'Collections',
  orders: 'Orders',
  customers: 'Customers',
  coupons: 'Coupons',
  inventory: 'Inventory',
  reviews: 'Reviews',
  content: 'Content',
  media: 'Media',
  analytics: 'Analytics',
  settings: 'Settings',
  newsletter: 'Newsletter',
  contact_messages: 'Messages',
};

File: backend/app/models/role.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, JSON, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.database import Base


ADMIN_MODULES = [
    "dashboard",
    "products",
    "categories",
    "collections",
    "orders",
    "customers",
    "coupons",
    "inventory",
    "reviews",
    "content",
    "media",
    "analytics",
    "settings",
    "newsletter",
    "contact_messages",
]


class Role(Base):
    __tablename__ = "roles"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(100), unique=True, nullable=False)
    slug = Column(String(100), unique=True, nullable=False, index=True)
    description = Column(Text, nullable=True)
    permissions = Column(JSON, nullable=False, default=list)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    users = relationship("User", back_populates="staff_role")
